src/atomate2/abinit/jobs/core.py

```diff
@@ -0,0 +1,253 @@
+        """Resolve dependencies to restart Non-Scf calculations.
+
+        Non-Scf calculations can only be restarted from the WFK file (or WFQ file).
+        """
+        # if self.restart_info.reset:
+        #     # remove non reset keys that may have been added in a previous restart
+        #     self.remove_restart_vars(["WFQ", "WFK"])
+        if True:
+            ext = self.restart_extension
+            prev_outdir = Directory(os.path.join(prev_dir, OUTDIR_NAME))
+            restart_file = prev_outdir.has_abiext(ext)
+            if not restart_file:
+                msg = f"Cannot find the {self.restart_extension} file to restart from."
+                logger.error(msg)
+                raise RestartError(msg)
+
+            # Move out --> in.
+            self.out_to_in(restart_file)
+
+            # Add the appropriate variable for restarting.
+            # Note that the restart of both WFK and WFQ is activated by irdwfk (i.e. ird var for WFK extension)
+            irdvars = irdvars_for_ext("WFK")
+            self.abinit_input_set.set_vars(irdvars)
+
+
+@dataclass
+class NonScfWfqMaker(NonScfMaker):
+    """Maker to create non SCF calculations for the WFQ."""
+
+    calc_type: str = "nscf_wfq"
+    name: str = "non-Scf calculation"
+
+    input_set_generator: NonScfWfqInputGenerator = NonScfWfqInputGenerator()
+    CRITICAL_EVENTS: Sequence[str] = ("NscfConvergenceWarning",)
+
+    # Here there is no way to set a default that is mutable (dict) for the dependencies so I use
+    # the default_factory of the dataclass field with a dict subclass...
+    # One option could be to use some kind of frozen dict/mapping.
+    # - frozendict (https://marco-sulla.github.io/python-frozendict/) is one possibility
+    #   but is not part of the stdlib. A PEP was actually discussed (and rejected) to
+    #   add such a frozendict in the stdlib (https://www.python.org/dev/peps/pep-0416/).
+    # - there is an ongoing PEP being discussed for a frozenmap type in the collections module.
+    #   If this PEP (https://www.python.org/dev/peps/pep-0603/) is accepted in the future, we
+    #   may think to use this new frozenmap type.
+    # Another option would be to use a different structure, e.g. a tuple of tuples. In the current
+    # case, this would give : ("scf", ("DEN", )). This is essentially a user defined mapping of course...
+    dependencies: Optional[dict] = field(default_factory=NonScfDeps)
+
+    wfq_tolwfr: float = 1.0e-22
+
+    # Non dataclass variables:
+    restart_extension = "WFQ"
+
+
+@dataclass
+class RelaxMaker(BaseAbinitMaker):
+    """Maker to create relaxation calculations."""
+
+    calc_type: str = "relax"
+    input_set_generator: RelaxSetGenerator = RelaxSetGenerator()
+    name: str = "Relaxation calculation"
+
+    CRITICAL_EVENTS: Sequence[str] = ("RelaxConvergenceWarning",)
+
+    # non-dataclass variables
+    structure_fixed: ClassVar[bool] = False
+
+    def resolve_restart_deps(self, prev_dir):
```

Comment:
Can we chat about the `resolve_restart_deps` function. I'm not entirely sure what is going on and maybe there is a cleaner way to handle it?

Reply:
Refactored and integrated inside the sets.